apps/server/src/external/pcglAuthZClient.ts

```diff
@@ -0,0 +1,137 @@
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { Request } from 'express';
+
+import { InternalServerError, ForbiddenError, NotFoundError } from '@overture-stack/lectern-dictionary';
+
+import { UserDataResponseErrorType } from '../common/types/auth';
+import { Groups, userDataResponseSchema, UserDataResponseSchemaType } from '../common/validation/auth-validation';
+import { authConfig } from '../config/authConfig';
+import logger from '../config/logger';
+import urlJoin from 'url-join';
+
+/**
+ *  Function to perform fetch requests to AUTHZ service
+ *
+ * @param resource endpoint to query from authz
+ * @param token authorization token
+ * @param options optional additional request configurations for the fetch call
+ *
+ */
+const fetchAuthZResource = async (resource: string, token: string, options?: RequestInit) => {
+	const { AUTHZ_ENDPOINT } = authConfig;
+
+	const url = urlJoin(AUTHZ_ENDPOINT, resource);
+	const headers = new Headers({
+		Authorization: `Bearer ${token}`,
+		'Content-Type': 'application/json',
+	});
+
+	try {
+		return await fetch(url, { headers, ...options });
+	} catch (error) {
+		logger.error(`[AUTHZ]: Something went wrong fetching authz service. ${error}`);
+		throw new InternalServerError(`Bad request: Something went wrong fetching from authz service`);
+	}
+};
+
+/**
+ * @param token Access token from Authz
+ * @returns validated object of UserDataResponse
+ */
+export const fetchUserData = async (token: string) => {
+	const response = await fetchAuthZResource(`/user/me`, token);
+
+	if (!response.ok) {
+		const errorResponse: UserDataResponseErrorType = await response.json();
+
+		logger.error(`[AUTHZ]: Unable to verify user response from AUTHZ. ${errorResponse}`);
+
+		const responseMessage =
+			'Something went wrong while verifying PCGL user account information, please try again later.';
+
+		switch (response.status) {
+			case 401:
+			case 403:
+				throw new ForbiddenError(responseMessage);
+			case 404:
+				throw new NotFoundError(
+					"This account is currently not associated within the PCGL project. This may be due to the fact that you haven't completed the onboarding process for new accounts, or have logged in with an account not previously used to access the service.",
```

Comment:
Let's simplify our error response message to describe the issue and not the potential causes.


```suggestion
					"This account is currently not associated within the PCGL project.",
```